src/main/java/com/bookshop/mapper/CartItemMapper.java

```diff
@@ -0,0 +1,24 @@
+package com.bookshop.mapper;
+
+import com.bookshop.config.MapperConfig;
+import com.bookshop.dto.cartitem.CartItemResponseDto;
+import com.bookshop.dto.cartitem.CreateCartItemDto;
+import com.bookshop.dto.cartitem.UpdateCartItemDto;
+import com.bookshop.model.CartItem;
+import com.bookshop.service.BookService;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+
+@Mapper(config = MapperConfig.class,
+        uses = BookService.class)
+public interface CartItemMapper {
+    @Mapping(target = "bookId", source = "book.id")
+    @Mapping(target = "bookTitle", source = "book.title")
+    CartItemResponseDto toResponseDto(CartItem cartItem);
+
+    @Mapping(source = "bookId", target = "book")
```

Comment:
how this mapping is supposed to work? you're trying to map `Long` to `Book`

correct me if I'm wrong

Reply:
it works because in the @Mapper annotation I used "uses = BookService.class" and this service has method "Book findBookById(Long id);"

Reply:
![mapper](https://github.com/TetianaDenchyk/online-bookshop/assets/104452767/2e546a55-9b45-4329-b6e3-c9229bb21b30)